### DISCUSSION TITLE
Extending mdx-editor / lexical editor

### DISCUSSION THREAD

**Discussion by jannikbuschke** [mdx-editor/editor#506] [Q&A] [answered]
I am wondering what the general philosophy here is. Is it possible to apply lexical knowledge to extend mdx-editor, or can this only be done via the Gurx state manangement thing?

https://mdxeditor.dev/editor/docs/extending-the-editor

What would need to be done to use a lexical plugin? Is it possible or would one need to reimplement it with an "mdx editor plugin"?

**Comment by petyosi** [comment, marked as answer]
Lexical knowledge is a must if you want to build custom editors, for example. For other things (like support of new syntax), you need understanding of the mdast specifics. 

Pure lexical plugins can be integrated. For example, I'm using the `MarkdownShortcutsPlugin` [like this](https://github.com/sanuradhag/editor/blob/548dcf2ddbbbda7f4f0f816a12c34d9dc35bda0c/src/plugins/markdown-shortcut/index.tsx#L28-L42). However, a Lexical plugin "plugin" system is usually just a component that does a bunch of stuff with the editor in `useEffect`. It's hard to make composition / reusability of state like this, as it will usually lead to a mess of spaghetti hooks and context. That's why I came up with the Gurx patterns. 

Hope this makes sense.

**Reply by noteverseio** [reply]
Yes, the MDXEditor can be highly extensible and easily integrated with Lexical Plugin, I have tried floating toolbar, slash command, collaboration, autocomplete etc from lexical playground into my editor, all working fine so far.

**Reply by jannikbuschke** [reply]
@noteverseio can you give some example code of a collaboration setup?
I somehow cannot get it to work

**Reply by NehalDamania** [reply]
> llaboration, autocomplete etc from lexical playground into my editor, all working

@noteverseio I am trying to implement `@mentions` and `#hashtags` autocomplete. Any examples or suggestions of autocomplete would be greatly appreciated.

**Reply by rob-at-cortex** [reply]
does this help anyone? it works for me, but would be good to get a better view on it...
https://github.com/mdx-editor/editor/pull/535

**Reply by rob-at-cortex** [reply]
@noteverseio, did you get collaboration working with tables ? any examples without forking this repo ?

**Reply by muslimmuda15** [reply]
> Yes, the MDXEditor can be highly extensible and easily integrated with Lexical Plugin, I have tried floating toolbar, slash command, collaboration, autocomplete etc from lexical playground into my editor, all working fine so far.

Can you give me an example?
I try to working on that, but it seem difficult on me.
